app/src/main/java/com/android/sample/model/workout/BodyWeightWorkout.kt

```diff
@@ -2,16 +2,22 @@ package com.android.sample.model.workout
 
 // BodyWeight workout description
 class BodyWeightWorkout(
+    workoutId: String,
     name: String,
     description: String,
```

Comment:
The main problem I see with this use of unique ID, is that it is the responsability of the one creating the workout, to provide an actually unique ID. This means that in order to provide a unique ID, we must keep track of what ID have been used before, which isn't really convenient. 

I have a suggestion. Have a look at the [UUID class from Java](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/UUID.html)
*There's actually a Uuid class in Kotlin, but only for Kotlin 2.0 and more which we don't use. So we use the fact the Kotlin is fully interoperable with Java*  sdafafdsfsa

Basically, I think a correct use here would be to change BodyWeightWorkout  definition to

`Class BodyWeightWorkout(name: String, description: String, warmup: Boolean, val exercises : MutableList<BodyWeightExercise> = mutableListOf()) {
val workoutId : String = UUID.randomUUID().toString()
}`
This way we can still access the workoutId using for workout.workoutId (for a given BodyWeightWorkout).

I have another suggestion that may be overkill, but actually creating an Interface for every obejcts that must be uniquely identified, such as Workouts, Exercises, and UserAccount might be a good idea for clarity. 

Something simple like ```interface UniquelyIdentifiable {
    val UUID: String
}```

This would provide an easier API since we would always use .UUID when we want to retrieve one instead of somethingId

Reply:
We should discuss this Interface thing I think

Reply:
I your comment. But as these workout/exercises have to be stored into the firestore, can't we just ask firestore for a new uid as we did for the boot camp when creating a new todo ?

Reply:
I actually forgot that these are stored in firebase. So my whole comment is irrelevant, sorry

Reply:
No problem. Thank you for reviewing the code !